cdk-testsuite/tests/openshiftSanity.py

```diff
@@ -0,0 +1,195 @@
+'''
+Created on Jun 29, 2016
+
+@author: amit
+'''
+from avocado import Test
+from avocado import VERSION
+import imp
+import logging
+import os
+
+log = logging.getLogger("Openshift.Debug")
+
+class OpenshiftTests(Test):
+
+    def setUp(self):
+        '''
+        preconfiguring the test setup before running each test case
+        Arg:
+            self (object): Object of the current method
+        '''
+        if os.name == "posix":
+            os.chdir(self.params.get('path_linux'))
+            self.log.info(self.params.get('path_linux'))
+        elif os.name == "nt":
+            os.chdir(self.params.get('path_win'))
+            self.log.info(self.params.get('path_win'))
+        else:
+            os.chdir(self.params.get('path_mac'))
+            self.log.info(self.params.get('path_mac'))
+        global openshift
+        openshift = imp.load_source('openshift', self.params.get('openshift_lib_MODULE'))
+        self.log.info("###########################################################################################")
+        self.log.info(openshift.openshiftLibInfo(self))
+        self.log.info("Avocado version : %s" % VERSION)
+        self.log.info("###########################################################################################")
+           
+    def test_python_project(self):
+        '''
+        Runs sanity on openshift s2i python source
+            1. login
+            2. new project create
+            3. App deploy using the registry from config file
+            4. Validation for successful deployment
+            5. Routing
+            6. Pod status
+            7. project delete
+            8. logout
+        Args:
+            self (object): Object of the current method
+            openshift_URL (string) : ip and port of openshift server to be used
+            openshift_USERNAME (string): username of openshift web console to be used
+            openshift_PASSWORD (string): password of openshift web console to be used
+            openshift_python_PROJECT (string): name of the python project to be added to the openshift server
+            openshift_python_REGISTRY (string): python registry path/location
+            service_python_NAME (string): name of the python service to be exposed outside
+        '''
+        openshift.new_project(self, self.params.get('openshift_URL'), self.params.get('openshift_USERNAME'), 
```

Comment:
What is IMHO ?
assert statement is implemented in the library file. So in the failure case its directly reflects in the output and also in the output log